packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -0,0 +1,242 @@
+import {
+  Component,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  h,
+} from '@stencil/core';
+
+import {
+  months,
+  days,
+} from '../../../../react-components/src/helpers/options-for-select.js';
+
+@Component({
+  tag: 'va-date',
+  styleUrl: 'va-date.css',
+  shadow: true,
+})
+export class VaDate {
+  /**
+   * Render marker indicating field is required.
+   */
+  @Prop() required: boolean;
+
+  /**
+   * Label for the field.
+   */
+  @Prop() label!: string;
+
+  /**
+   * Used to create unique name attributes for each input.
+   */
+  @Prop() name!: string;
+
+  /**
+   * The error message to render (if any)
+   * This prop should be leveraged to display any custom validations needed for this component
+   */
+  @Prop() error: string;
+
+  /**
+   * Set the `min` value on the year input.
+   */
+  @Prop() minYear: number;
+
+  /**
+   * Set the `max` value on the year input.
+   */
+  @Prop() maxYear: number;
+
+  /**
+   * Set the default month on the month input.
+   */
+  @Prop() month: string;
+
+  /**
+   * Set the default day on the day input.
+   */
+  @Prop() day: string;
+
+  /**
+   * Set the default year on the year input.
+   */
+  @Prop() year: string;
```

Comment:
Can we also get a `value` prop since that's an established pattern for inputs? That would also let us use the familiar `event.target.value` pattern.